### PR TITLE
Replace some "assert_throws(new FooError(), stuff)" calls with assert_throws_js

### PR DESCRIPTION
**Pull request by stephenmcgruer**
This is an attempt to understand what part of #21350 is failing the CI; please excuse the email spam when wpt-pr-bot cc's you to this PR.

### CODE REVIEW

**Comment by stephenmcgruer** [comment]
Ah, I've figured out what was happening here @bzbarsky . You were (unwittingly) updating tests that are commit-time generated, without changing the generation logic. So we run a CI to check that they don't change, and re-running the generation undid your work and left the CI check with a diff (which fails it). I'll figure out how to update the generation logic.

**Comment by bzbarsky** [comment]
Aha!  I knew we had some pre-generated tests, but not ones that are commit-time generated....

File: 2dcontext/tools/gentestutils.py [hunk 1]
```diff
@@ -233,7 +233,7 @@ def expand_test_code(code):
                 code)
 
         code = re.sub(r'@assert throws (\S+Error) (.*);',
-                r'assert_throws(new \1(), function() { \2; });',
+                r'assert_throws_js(\1, function() { \2; });',
```
Comment (bzbarsky) [inline 2dcontext/tools/gentestutils.py]: The bit right above this probably needs to become `assert_throws_dom("\1", etc`, right?

Comment (stephenmcgruer) [reply 2dcontext/tools/gentestutils.py]: Yes, I will do that when I get to the commit where you did `assert_throws_dom` rewrites.

**Comment by stephenmcgruer** [comment]
> Aha! I knew we had some pre-generated tests, but not ones that are commit-time generated....

Me neither! It's why it took me far too long to realize what was going on :D

**Comment by stephenmcgruer** [comment]
Firefox stability timed out (expected for a large CL). Chrome stability had some flakes; might be pre-existing but I am a bit concerned because I can't find previous PRs for the same files that show the flake. Retrying the CI to see if the flake changes.

**Comment by stephenmcgruer** [comment]
Ok, so on the second run for Chrome stability, the flaky trusted-types/ failures disappeared, and the xhr/ failure remains:

```
65:21.88 INFO ## Unstable results ##
65:21.88 INFO |                  Test                  |                       Subtest                        |          Results           | Messages |
65:21.88 INFO |----------------------------------------|------------------------------------------------------|----------------------------|----------|
65:21.88 INFO | `/xhr/setrequestheader-bogus-name.htm` | `setRequestHeader should throw with header name "".` | **Duplicate subtest name** |          |
```

Looking at that test, it looks like that is an already-existing problem, so I'm not too worried about that.

In terms of reported results, [Safari](https://wpt.fyi/results/?diff&filter=ADC&run_id=393610007&run_id=416810013) and [Firefox](https://wpt.fyi/results/?diff&filter=ADC&run_id=395470012&run_id=391720018) are clear, no diff. [Chrome](https://wpt.fyi/results/?diff&filter=ADC&run_id=397680014&run_id=406970013) is showing a diff:

```
  | css/ | 5676 / 5997 | 5572 / 5997 | 0 / -104 / 0
  | kv-storage/ | 26 / 26 | 25 / 26 | 0 / -1 / 0
  | webrtc/ | 195 / 247 | 187 / 247 | 0 / -8 / 0
  | websockets/ | 64 / 180 | 60 / 180 | 0 / -4 / 0
```

New failures are:

- [css/css-properties-values-api/register-property-syntax-parsing.html](https://wpt.fyi/results/css/css-properties-values-api/register-property-syntax-parsing.html?diff&filter=ADC&run_id=397680014&run_id=406970013) : many
- [css/css-properties-values-api/register-property.html](https://wpt.fyi/results/css/css-properties-values-api/register-property.html?diff&filter=ADC&run_id=397680014&run_id=406970013) : two
- [css/css-typed-om/stylevalue-subclasses/numeric-objects/*](https://wpt.fyi/results/css/css-typed-om/stylevalue-subclasses/numeric-objects?diff&filter=ADC&run_id=397680014&run_id=406970013) : multiple tests
- [kv-storage/interface.https.html](https://wpt.fyi/results/kv-storage/interface.https.html?diff&filter=ADC&run_id=397680014&run_id=406970013) : Same-realm brand checks
- [webrtc/RTCConfiguration-iceServers.html](https://wpt.fyi/results/webrtc/RTCConfiguration-iceServers.html?diff&filter=ADC&run_id=397680014&run_id=406970013) : multiple, regarding SyntaxError
- [websockets/stream-tentative/*](https://wpt.fyi/results/websockets/stream-tentative?diff&filter=ADC&run_id=397680014&run_id=406970013) :  constructing with an invalid URL should throw

These all look valid and I've reproduced at least one locally. It's possible that these tests would work as expected if all of the changes in the original PR were applied together. For now, I am **reverting** the changes to those files, and we can look at adding them back later.

**Comment by stephenmcgruer** [comment]
Ok, updated the commit with the following files reverted:

```
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	modified:   css/css-properties-values-api/register-property-syntax-parsing.html
	modified:   css/css-properties-values-api/register-property.html
	modified:   css/css-typed-om/stylevalue-subclasses/numeric-objects/cssMathValue.tentative.html
	modified:   css/css-typed-om/stylevalue-subclasses/numeric-objects/parse.tentative.html
	modified:   css/css-typed-om/stylevalue-subclasses/numeric-objects/to.tentative.html
	modified:   css/css-typed-om/stylevalue-subclasses/numeric-objects/toSum.tentative.html
	modified:   kv-storage/interface.https.html
	modified:   webrtc/RTCConfiguration-iceServers.html
	modified:   websockets/stream-tentative/constructor.any.js
```

Lets see how this does through the CI :)

**Comment by bzbarsky** [comment]
> It's possible that these tests would work as expected if all of the changes in the original PR were applied together.

No, that shouldn't matter.

Looking at those tests,:

1) `css/css-properties-values-api/register-property-syntax-parsing.html` -- this should have `assert_throws_dom("SyntaxError", ...)`.  It was using the wrong thing before, and since this test is generally failing in Firefox I didn't catch this part.
2) `css/css-properties-values-api/register-property.html` -- same thing.  I should probably have looked for `new SyntaxError()` explicitly and fixed those up separately....
3) `css/css-typed-om/stylevalue-subclasses/numeric-objects/cssMathValue.tentative.html` -- same thing.
4) `css/css-typed-om/stylevalue-subclasses/numeric-objects/parse.tentative.html` -- same thing.
5) `css/css-typed-om/stylevalue-subclasses/numeric-objects/to.tentative.html` -- same thing.
6) `css/css-typed-om/stylevalue-subclasses/numeric-objects/toSum.tentative.html` -- same thing.
7)  `kv-storage/interface.https.html` -- the "Same-realm brand checks" test should be using `iframe.contentWindow.TypeError` throughout.
8)  `webrtc/RTCConfiguration-iceServers.html`-- this is the `SyntaxError` thing.
9) `websockets/stream-tentative/constructor.any.js` -- this is the `SyntaxError` thing.

Let's see how the rest of this stuff goes, and then I can do a separate PR to fix these tests properly.  Thank you again for sorting through this!

**Comment by stephenmcgruer** [comment]
Stability checks fail as expected, but results are good for all browsers (no diffs). Merging (this was reviewed in https://github.com/web-platform-tests/wpt/pull/21350)

**Comment by bzbarsky** [comment]
> and then I can do a separate PR to fix these tests properly

https://github.com/web-platform-tests/wpt/pull/21380

**Pull request merged**